### ISSUE TITLE
missing "aws_iam_role" documentation

### ISSUE THREAD

**Issue by JeanMertz** [hashicorp/terraform#1882]
I noticed the new AWS IAM support metioned in the documentation, but it seems that

1) the documentation of IAM resources are somewhat lacking, with a description like "Provides an IAM role policy." not really helping make clear what this resource does exactly. I know I can open up the AWS documentation (and I did), but given that other resources have a more in-depth description, these ones seem to be lacking.

2) the `aws_iam_role` documentation seems to be missing.

I haven't had the time yet to provide a PR for this, so this issue will have to do for now.

**Comment by JeanMertz** [comment]
Also, this page is incorrect: https://www.terraform.io/docs/providers/aws/r/iam_instance_profile.html

It states that `aws_iam_role` has a `policy` attribute, which it does not. The correct attribute is `assume_role_policy`. But simply swithing those is not enough, because a `AssumeRole` policy is required for this to work:

```
* Error creating IAM Role describe: MalformedPolicyDocument: AssumeRole policy must contain principals.
```

**Comment by JeanMertz** [comment]
Also, the `aws_instance#iam_instance_profile` attribute should probably be updated to state that you need to pass in the name attribute of the profile, instead of the ID (which is probably the default expectation and the way most references work in Terraform).

**Comment by erikthorselius** [comment]
:+1:

**Comment by radeksimko** [comment]
Fixed in #1935
